src/FakeItEasy/Creation/FakeObjectCreator.cs

```diff
@@ -66,10 +66,23 @@ internal class FakeObjectCreator : IFakeObjectCreator, IMethodInterceptionValida
         }
 
         [SuppressMessage("Microsoft.Design", "CA1021:AvoidOutParameters", MessageId = "2#", Justification = "Seems appropriate here.")]
-        public bool MethodCanBeInterceptedOnInstance(MethodInfo method, object? callTarget, [NotNullWhen(false)]out string? failReason) =>
-            callTarget is object && DelegateCreationStrategy.IsResponsibleForCreating(callTarget.GetType())
-                ? this.delegateCreationStrategy.MethodCanBeInterceptedOnInstance(method, callTarget, out failReason)
-                : this.defaultCreationStrategy.MethodCanBeInterceptedOnInstance(method, callTarget, out failReason);
+        public bool MethodCanBeInterceptedOnInstance(MethodInfo method, object? callTarget, [NotNullWhen(false)] out string? failReason)
+        {
+            if (callTarget is object)
+            {
+                if (DelegateCreationStrategy.IsResponsibleForCreating(callTarget.GetType()))
+                {
+                    return this.delegateCreationStrategy.MethodCanBeInterceptedOnInstance(method, callTarget, out failReason);
+                }
+                else
+                {
+                    return this.defaultCreationStrategy.MethodCanBeInterceptedOnInstance(method, callTarget, out failReason);
+                }
+            }
+
+            failReason = "Call target is null";
+            return false;
+        }

```

Comment:
There was a CS8762 warning, because `failReason` ended up being null when `callTarget` was null, violating the `[NotNullWhen(false)]` contract.

The conditional expression *should* have worked, but produced the same warning. I opened https://github.com/dotnet/roslyn/issues/44886

Reply:
> There was a CS8762 warning, because `failReason` ended up being `null` when `callTarget` was null, violating the `[NotNullWhen(false)]` contract.

I think this comment was found to be false, right? Not complaining, just making sure I understand.

Reply:
> I think this comment was found to be false, right?

Yes, `callTarget` being null had nothing to do with it. It's just the compiler bug, after all.